### PR TITLE
Pingdemic Screen

### PR DESCRIPTION
**Pull request by bionara**
# Description

Initial work on "pingdemic" front end. Getting PR up for initial review before EOP.
Run locally against this backend: https://github.com/zoe/covid-app/pull/529
- or edit ~line 60 on HowYouFeelScreen to navigate and return without need to do checks (this will let you see/test the UI without the flag setting).

Have reused the covid test date/range picker, which has "yes/no" buttons vs slider (see screen shot)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots
image
image
image

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: assets/lang/en.json [hunk 1]
```diff
@@ -439,6 +439,32 @@
     "vaccine-reporting-link": "Why is that important?"
   },
 
+  "pingdemic": {
+    "title": "Understanding the COVID risk profile of people asked to self-isolate",
+    "body": "We're looking to understand the COVID risk profile of people contacted by the NHS COVID-19 App to self-isolate compared to those who have not been.",
+    "q1": "In the last month, were you asked by the NHS Covid-19 app to self-isolate based on the app detecting that you were in contact with someone who has COVID-19?",
```
Comment (juliatan) [inline assets/lang/en.json:445]: COVID-19 instead of Covid-19 for consistency?

File: assets/lang/es.json [hunk 2]
```diff
@@ -436,6 +436,32 @@
     "vaccine-reporting-link": ""
   },
 
+  "pingdemic": {
+    "title": "Understanding the COVID risk profile of people asked to self-isolate",
+    "body": "We're looking to understand the COVID risk profile of people contacted by the NHS COVID-19 App to self-isolate compared to those who have not been.",
+    "q1": "In the last month, were you asked by the NHS Covid-19 app to self-isolate based on the app detecting that you were in contact with someone who has COVID-19?",
+    "q1-yes": "Yes  > Select date",
```
Comment (juliatan) [inline assets/lang/es.json:443]: Not sure the part of ">" should be shown?

File: assets/lang/es.json [hunk 3]
```diff
@@ -436,6 +436,32 @@
...
+    "q1-no": "No > do you have the app?",
```
Comment (juliatan) [inline assets/lang/es.json:444]: Ditto above, and also for line below.

File: assets/lang/es.json [hunk 4]
```diff
@@ -436,6 +436,32 @@
...
+    "q1-prefer-not-to-say": "Prefer not to say  > do you have the app?",
+    "risk-title": "Understanding the COVID risk profile of people asked to self-isolate",
+    "risk-body": "We're looking to understand the COVID risk profile of people contacted by the NHS COVID-19 App to self-isolate compared to those who have not been.",
+    "risk-q1": "In the last month, were you asked by the NHS Covid-19 app to self-isolate based on the app detecting that you were in contact with someone who has COVID-19?",
```
Comment (juliatan) [inline assets/lang/es.json:448]: Covid-19 vs COVID-19

File: assets/lang/es.json [hunk 5]
```diff
@@ -436,6 +436,32 @@
...
+    "yes": "Yes",
+    "no": "No",
+    "prefer-not-to-say": "Prefer not to say",
+    "dates": "What date were you asked to self-isolate from? (Optional)",
```
Comment (juliatan) [inline assets/lang/es.json:452]: Should we be using the rest of the app convention in that non-optional questions are indicated with "*", and everything else is optional?

File: src/features/pingdemic/PingdemicScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+import { CovidTestDateQuestion, ICovidTestDateData } from '../covid-tests/fields';
+import { BrandedButton } from '@covid/components';
+import { PingdemicApiClient } from './PingdemicApiClient';
+import { PingdemicRequest } from './PingdemicRequest';
+import moment from 'moment';
+import NavigatorService from '@covid/NavigatorService';
+
+const pingdemicApiClient = new PingdemicApiClient();
+
+type Item = {
```
Comment (juliatan) [inline src/features/pingdemic/PingdemicScreen.tsx:23]: Rename to `tItem`?

File: src/features/pingdemic/PingdemicRequest.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export type PingdemicRequest = {
+  patient?: string; //	Patient id
```
Comment (juliatan) [inline src/features/pingdemic/PingdemicRequest.ts:2]: Is this really optional?

File: src/features/pingdemic/PingdemicScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
+import { RadioInput } from '@covid/components/inputs/RadioInput';
+import Screen, { Header } from '@covid/components/Screen';
+import { HeaderText, RegularText } from '@covid/components/Text';
+import { assessmentCoordinator } from '@covid/core/assessment/AssessmentCoordinator';
+import { ScreenParamList } from '@covid/features';
+import i18n from '@covid/locale/i18n';
+import { RouteProp, useIsFocused } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { Formik, FormikProps } from 'formik';
+import { Form } from 'native-base';
```
Comment (juliatan) [inline src/features/pingdemic/PingdemicScreen.tsx:10]: Should we be using FormWrapper?

**Review by juliatan — COMMENTED** [review]
Hey Ben, noted a few comments here. I need to jet, so haven't pulled down your branch to do a proper check but can do so tmw.

File: src/features/pingdemic/PingdemicScreen.tsx (hunk 6 above)
Comment (bionara) [reply src/features/pingdemic/PingdemicScreen.tsx:22]: done

File: src/features/pingdemic/PingdemicScreen.tsx (hunk 8 above)
Comment (bionara) [reply src/features/pingdemic/PingdemicScreen.tsx:10]: updated!

File: assets/lang/es.json (hunk 2 above)
Comment (bionara) [reply assets/lang/es.json]: Cheers, somesh did an overhaul on all text!

File: src/features/pingdemic/PingdemicScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+};
+
+interface IPingdemicData extends ICovidTestDateData {
+  askedByApp?: string;
+  haveApp?: string;
+  appActiveWithBluetooth?: string;
+}
+
+type Props = {
+  navigation: StackNavigationProp<ScreenParamList, 'HowYouFeel'>;
```
Comment (aikewoody) [inline src/features/pingdemic/PingdemicScreen.tsx:35]: Should this be `Pingdemic`?

File: src/features/register/CountryIpModal.tsx [hunk 10]
```diff
@@ -24,7 +24,7 @@ interface IProps extends ITest {
   navigation: StackNavigationProp<ScreenParamList, 'Welcome'>;
 }
 
-type Item = {
+type tItem = {
```
Comment (aikewoody) [inline src/features/register/CountryIpModal.tsx:27]: `TItem`with capitalised `T` ;)

**Review by aikewoody — APPROVED** [review]

**Review by selvach — DISMISSED** [review]
Amazing!

File: src/features/pingdemic/PingdemicScreen.tsx (hunk 9 above)
Comment (selvach) [reply src/features/pingdemic/PingdemicScreen.tsx]: Fixed! Great spot!

**Review by selvach — APPROVED** [review]

**Pull request merged**